### PR TITLE
executor: rewrite TestNotFillCacheFlag to avoid data race in CI

### PR DESCRIPTION
**Pull request by tiancaiamao**
The old test use a user-defined client to check request, but data race is detected in  `checkRequestClient.notFillCache`
`checkRequestClient` is a global instance used by all goroutine, that's bad.

I rewrite this test case to use `CheckSelectRequestHook` so the CI could be more stable.

@XuHuaiyu @winkyao

### CODE REVIEW

File: executor/executor_test.go [hunk 1]
```diff
@@ -2251,27 +2245,32 @@ func (s *testContextOptionSuite) TestCoprocessorPriority(c *C) {
...
+		sql    string
+		expect bool
+	}{
+		{"select SQL_NO_CACHE * from t", true},
+		{"select SQL_CACHE * from t", false},
+		{"select * from t", false},
+	}
+	ctx := context.Background()
+	for _, test := range tests {
+		ctx1 := context.WithValue(ctx, "CheckSelectRequestHook", func(req *kv.Request) {
```
Comment (shenli) [inline executor/executor_test.go]: Where is this hook used?

Comment (tiancaiamao) [reply executor/executor_test.go]: It's introduced here https://github.com/pingcap/tidb/pull/5973/files#diff-76317381b0cf3cac2705ddf925c838b2R313

**Pull request merged**